Database driven organizers page

davidfischer opened the pull request:
- Adds the first migration to the app
- Adds an Organizer model that will be displayed on the organizer page
- This PR does not yet display images but it would be great if profile photos came from Meetup profiles.
- Should these organizers have a brief description?

Replaces #155. 
Fixes #64.

## Screenshot
![Screenshot from 2024-05-30 23-03-54](https://github.com/sandiegopython/pythonsd-django/assets/185043/428d6d03-57d5-44c4-a5c2-2ed1e7f5c7ec)

## Deploying
This PR adds our first migration needed for the site. As with any kind of migration, after this is merged, somebody (probably me) will need to run this migration on the production database.

Jer-Pha committed:
feat: add database-driven organizer page

davidfischer committed:
Fixup the organizers PR

- Add code migration
- Fix linting
- Remove inactive organizers

Jer-Pha commented:
Here's a CSV to easily copy/paste the names and Meetup URLs in /admin/. I don't know who is considered active so I sorted the list based on "Last visited" on Meetup. 

[pythonsd-organizers.csv](https://github.com/sandiegopython/pythonsd-django/files/15493392/pythonsd-organizers.csv)

davidfischer committed:
Handle uploading organizer photos

Jer-Pha commented on pythonsd/templates/pythonsd/organizers.html:
```diff
@@ -0,0 +1,48 @@
+      <img src="{{ organizer.photo.url }}" class="card-img-top" alt="{{ organizer.name }}">
+      <div class="card-body">
+        <h5 class="card-title">{{ organizer.name }}</h5>
+        <p class="card-text">{{ organizer.description }}</p>
```
What is `organizer.description` referencing?

davidfischer replied:
Heh. Nothing. I thought it might be a good idea but decided maybe it wasn't necessary. Do you have any thoughts there or should we cut it?

Jer-Pha replied:
I think it comes down to how much information is necessary here vs. what's displayed at the linked profiles.

A quick blurb about the organizer could be a nice touch, but where would it come from? Pulled from one of the profiles? Written by the organizer? How often would it be updated? 

Another option could be a start date. This could be displayed as something like "Organizer since YEAR" or "Contributing for _ years" (using timesince).

I'm not sure if any extra information is necessary since there will always be more info when clicking through to a profile. On the other hand, the card format definitely allows for more information at a glance.

davidfischer replied:
The blurb could come from the database but it's probably overkill. I'll just cut it for now and we can add one later if it feels useful.

davidfischer commented on pythonsd/templates/pythonsd/organizers.html:
```diff
@@ -0,0 +1,48 @@
+      <img src="{{ organizer.photo.url }}" class="card-img-top" alt="{{ organizer.name }}">
+      <div class="card-body">
+        <h5 class="card-title">{{ organizer.name }}</h5>
+        <p class="card-text">{{ organizer.description }}</p>
```
```suggestion

```

davidfischer replied:
```suggestion
```

davidfischer committed:
Update pythonsd/templates/pythonsd/organizers.html

Remove organizer description for now

Jer-Pha commented on pythonsd/models.py:
```diff
@@ -0,0 +1,22 @@
+        help_text="Recommended size of 400*400px or larger square",
+    )
+
+    def __str__(self):
```
```suggestion
    class Meta:
        indexes = [
            models.Index(
                fields=["active", "name"], 
                name="organizer_active_name_idx",
            ),
        ]

    def __str__(self):
```

This might be a bit overkill for how few organizers there are, but this is what I would do if the QuerySet was larger.

This would also need to be changed in `0001_initial.py`

davidfischer replied:
I hope you don't mind, but I'll probably skip this. I think we can index just "active" if necessary but I'm expecting the table to be tiny. Normally, I'd probably just index "active" and rely on Postgres' ability to use multiple indexes rather than creating a composite index (which you'd do for MySQL unless my knowledge is outdated).

Jer-Pha replied:
I definitely don't mind, just a premature optimization suggestion.

davidfischer committed:
Use docker compose for debugging production

- Add a docker compose environment with Redis and Postgres
- Add django-storages for storing media files in S3/R2

davidfischer commented:
I created a separate PR (#158) that merges into this one. That PR sets up Docker Compose which I really wanted in order to test this PR with Postgres and a custom storage backend. The custom media storage backend is what I'm planning to use to store organizer photos in something like S3 (or more specifically [Cloudflare R2](https://developers.cloudflare.com/r2/)). This PR I think is ready to review/merge and it could be merged without merging the other.

davidfischer committed:
Fixes around using a custom domain with S3/R2

- Set both `MEDIA_URL` and `AWS_S3_CUSTOM_DOMAIN`.
  This ensures that both using `MEDIA_URL` or
  using `file.url` work well

Jer-Pha reviewed (approved):
Looks good.

davidfischer commented:
> Looks good.

Thanks for the review. I asked in discord but I'll probably merge #158 today or tomorrow and deploy this. I don't want it to sit around too long.

Jer-Pha commented:
I looked over the changes in #158 but don't know enough about the non-Django stuff, which is why I only mentioned it in #154. I started my Docker journey this weekend, so I'll hopefully be more helpful with future PRs.

davidfischer committed:
Merge pull request #158 from sandiegopython/davidfischer/docker-compose-media-storage

Use docker compose for debugging production
